src/Compilers/Server/ServerShared/DiagnosticListener.cs

```diff
@@ -16,10 +16,15 @@ internal interface IDiagnosticListener
         void UpdateKeepAlive(TimeSpan timeSpan);
 
         /// <summary>
+        /// Called when a connection to the server occurs.
+        /// </summary>
+        void Connection();
```

Comment:
This type of usage is a fuzzy boundary.  I agree `event` or `interface` could work here.  I don't have a particularly good justification for choosing `interface` here and could be convinced to change it.  Probably in a separate change though.